app/controllers/repositories_controller.rb

```diff
@@ -198,12 +198,32 @@ def show
   end
 
   def create
-    @client = Client.new(safe_params)
+    prefix = nil
 
+    @client = Client.new(safe_params)
     authorize! :create, @client
 
-    if @client.save
+    provider_prefix = @client.provider.provider_prefixes.select { |_provider_prefix| _provider_prefix.state == "without-repository" }.first
+
+    if !provider_prefix.present?
+      prefix = Prefix.all.select { |_prefix| (_prefix.state == "unassigned") }.first
+    end
+
+    if (provider_prefix.present? || prefix.present?) && @client.save
+      if !provider_prefix.present?
+        provider_prefix = ProviderPrefix.new({ "provider_id": @client.provider.symbol, "prefix_id": prefix.uid })
+        provider_prefix.save
+      end
+
+      client_prefix = ClientPrefix.new(
+        client_id: @client.symbol,
+        provider_prefix_id: provider_prefix.uid,
+        prefix_id: provider_prefix.prefix.uid,
+      )
+      client_prefix.save
```

Comment:
I do think that more test for the controller would be needed. https://github.com/datacite/lupo/blob/master/spec/requests/clients_spec.rb